modules/synthetics/main.tf

```diff
@@ -4,16 +4,23 @@ locals {
   enabled            = module.this.enabled
   alert_tags         = local.enabled && var.alert_tags != null ? format("%s%s", var.alert_tags_separator, join(var.alert_tags_separator, var.alert_tags)) : ""
   datadog_synthetics = { for k, v in var.datadog_synthetics : k => v if local.enabled }
+
+  all_public_locations = sort(keys(data.datadog_synthetics_locations.public_locations.locations))
+  locations            = contains(split(",", lower(join(",", var.locations))), "all") ? local.all_public_locations : var.locations
+
```

Comment:
this somehow resolves to var.locations instead of the data sources result and so terratest fails because it doesnt recognize "all" as a proper value to feed to the datadog api

Reply:
This should be because I was using var.locations - which is fine if we pass it in via the component, but since in the test I'm defining it as part of the synthetic, we need to use `each.value.locations` which I've now updated it to use